aineko/core/dataset.py

```diff
@@ -153,12 +154,19 @@ def _update_offset_to_latest(self) -> None:
             partitions = self.consumer.assignment()
 
         for partition in partitions:
-            partition.offset = (
-                self.consumer.get_watermark_offsets(
-                    partition, cached=self.cached
-                )[1]
-                - 1
-            )
+            high_offset = self.consumer.get_watermark_offsets(
+                partition, cached=self.cached
+            )[1]
```

Comment:
is there any chance this line could fail? for example, if get_watermark_offsets didn't return a tuple, then the [1] would error. if the method is strict in what it can return, then it should be fine

Reply:
We should handle this upstream in `_update_offset_to_latest`. If this function fails because `get_watermark_offsets()` returns a `KafkaException` then I think we could either gracefully fail by using the same offset as the last call or raise a fatal error.